src/deserialize.rs

```diff
@@ -218,9 +220,9 @@ where &'a dyn Array: FromArrow<T>
     }
 }
 
-/// Returns an iterator for elements from a [`arrow2::array::Array`]
+/// Internal helper to return an iterator for elements from a [`arrow2::array::Array`]. 
 /// Does not perform any schema checks.
-pub fn arrow_array_deserialize_unchecked_iter<'a, T>(b: &'a dyn arrow2::array::Array) -> arrow2::error::Result<impl Iterator<Item = T> + 'a>
+fn arrow_array_deserialize_iter<'a, T>(b: &'a dyn arrow2::array::Array) -> arrow2::error::Result<impl Iterator<Item = T> + 'a>
```

Comment:
@jorgecarleitao I couldn't think of a better name, so I removed the "unchecked" and made this private for now (since it's only used internal to this file). We can come up with a better name if this becomes a public API. I think we should probably check the schema for all public methods anyways.